### PR TITLE
Refactor updates [API-1770] [API-1774]

### PR DESCRIPTION
**Pull request by yuce**

### CODE REVIEW

File: base/commands/config/config_add.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+func (cm AddCmd) Init(cc plug.InitContext) error {
+	cc.SetCommandUsage("add TARGET")
+	short := "Adds a configuration"
+	long := "Adds a configuration with the given name/path and KEY=VALUE pairs"
+	cc.SetCommandHelp(long, short)
+	cc.SetPositionalArgCount(1, math.MaxInt)
+	return nil
+}
+
+func (cm AddCmd) Exec(ctx context.Context, ec plug.ExecContext) error {
```
Comment (harunalpak) [inline base/commands/config/config_add.go:29]: ctx is unused parameter. Can we use _ for this?

File: base/commands/config/config_add.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		ps := strings.SplitN(arg, "=", 2)
+		if len(ps) != 2 {
+			return fmt.Errorf("invalid key=value pair: %s", arg)
+		}
+		opts = append(opts, clc.KeyValue[string, string]{
+			Key:   ps[0],
+			Value: ps[1],
+		})
+	}
+	dir, cfgPath, err := config.Create(target, opts)
```
[the remark this answers is not in the document]
Comment (harunalpak) [reply base/commands/config/config_add.go:42]: May be firstly we read values from files then add or override key/value pairs to configuration can be good solution.

File: base/commands/map/map_set.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	if _, err := ec.Props().GetBlocking(mapPropertyName); err != nil {
+		return err
+	}
+	keyStr := ec.Args()[0]
+	valueStr := ec.Args()[1]
+	kd, vd, err := makeKeyValueData(ec, ci, keyStr, valueStr)
+	if err != nil {
+		return err
+	}
+	ttl := GetTTL(ec)
```
Comment (harunalpak) [inline base/commands/map/map_set.go:46]: When I check it ttl values and maxIdle values could not get correctly.

Comment (yuce) [reply base/commands/map/map_set.go:46]: Thanks for catching that, fixed at: https://github.com/hazelcast/hazelcast-commandline-client/pull/171/commits/bdd411ae442c264ab8ec898e973972356335353d

**Comment by harunalpak** [comment]
When I run test, it fails. Can you check test please?

```
 Error Trace:    /Users/harun.alpak/Documents/Projects/hazelcast-commandline-client/clc/config/config_test.go:205
                Error:          Not equal: 
                                expected: "key1: value1\nkey2: value2\ncluster:\n  name: pr-3814\nssl:\n  ca-path: ca.pem\n  cert-path: cert.pem\n"
                                actual  : "key1: value1\nkey2: value2\nssl:\n  ca-path: ca.pem\n  cert-path: cert.pem\ncluster:\n  name: pr-3814\n"
```

File: base/commands/config/config_import.go [hunk 4]
```diff
@@ -149,86 +157,114 @@ func (cm ImportCmd) makeConfigFromZip(ctx context.Context, ec plug.ExecContext,
...
-		}
+		// find the configuration bits
+		token, clusterName, pw, cfgFound := extractConfigFields(reader, goPaths)
 		if !cfgFound {
 			return nil, errors.New("go file with configuration not found")
 		}
-		return nil, nil
+		opts := makeViridianOpts(clusterName, token, pw)
+		path = target
+		if target == "" {
```
[the remark this answers is not in the document]
Comment (yuce) [reply base/commands/config/config_import.go]: I've updated the help to reflect the current behavior at: https://github.com/hazelcast/hazelcast-commandline-client/pull/171/commits/4f3ef69e198f6a0a45ff80a90627c33291f72444

File: base/commands/config/config_add.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuce) [reply base/commands/config/config_add.go:45]: I've updated the help to reflect the current behavior at: https://github.com/hazelcast/hazelcast-commandline-client/pull/171/commits/4f3ef69e198f6a0a45ff80a90627c33291f72444

File: base/commands/config/config_import.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (yuce) [reply base/commands/config/config_import.go]: Removed at: https://github.com/hazelcast/hazelcast-commandline-client/pull/171/commits/4f3ef69e198f6a0a45ff80a90627c33291f72444

File: base/commands/config/config_add.go (hunk 1 above)
Comment (yuce) [reply base/commands/config/config_add.go]: Updated at: https://github.com/hazelcast/hazelcast-commandline-client/pull/171/commits/4f3ef69e198f6a0a45ff80a90627c33291f72444

File: clc/config/config.go [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+}
+
+func copySection(name string, level int, sb *strings.Builder, opts clc.KeyValues[string, string]) {
+	if len(opts) == 0 {
+		return
+	}
+	var leaves clc.KeyValues[string, string]
+	var sect clc.KeyValues[string, string]
+	sub := map[string]clc.KeyValues[string, string]{}
+	for _, opt := range opts {
```
Comment (harunalpak) [inline clc/config/config.go:199]: Can we check for possible options? 
For example I can add config with unrelated configuration options.
For example;

```
config add aliveli/pr-621 aaa=bbb ccc=dddd
aaa: bbb
ccc: dddd
```

Comment (harunalpak) [reply clc/config/config.go:199]: And also when I try the following command;

```
config add aliveli/pr-621 aaa.bbb.ccc.ddd='bbb' ccc.bbb='dddd'
Created yaml is;
aaa:
  bbb.ccc.ddd:
    ccc.ddd:
      ddd: bbb
ccc:
  bbb: dddd

It should be;
aaa:
  bbb:
    ccc:
      ddd: bbb
ccc:
  bbb: dddd
```

**Comment by yuce** [comment]
> When I run test, it fails. Can you check test please?

Sorted the configuration options at: https://github.com/hazelcast/hazelcast-commandline-client/pull/171/commits/cceadd203e323dff6937962b7f48701e85b87349
https://github.com/hazelcast/hazelcast-commandline-client/pull/171/commits/cceadd203e323dff6937962b7f48701e85b87349 so the config order is fixed.

File: clc/config/config.go (hunk 5 above)
Comment (yuce) [reply clc/config/config.go:203]: It's OK to add any option, since a plugins may use their own configuration options.

**Review by harunalpak — APPROVED** [review]

**Pull request merged**